sdk/helper/certutil/helpers.go

```diff
@@ -586,21 +594,32 @@ func ValidateSignatureLength(hashBits int) error {
 	return nil
 }
 
-func ValidateKeyTypeLength(keyType string, keyBits int) error {
+func ValidateKeyTypeLength(keyType string, keyBits *int) error {
+	if *keyBits == 0 {
+		newValue, present := defaultAlgorithmKeyBits[keyType]
+		if present {
+			*keyBits = newValue
```

Comment:
I'll defer this to @sgmiller, but I'm not a huge fan of overwriting pointer values within functions like this. I'd rather see functions returning the proper values to use. I understand that will require more refactoring so I am not sure if it's worth the effort.

Reply:
And of course nvm, Scott already mentioned in while I was busy reviewing...